src/Instrumentation/PDO/src/PDOInstrumentation.php

```diff
@@ -0,0 +1,167 @@
+
+        hook(
+            \PDO::class,
+            '__construct',
+            pre: static function (\PDO $pdo, array $params, string $class, string $function, ?string $filename, ?int $lineno) use ($instrumentation) {
+                /** @psalm-suppress ArgumentTypeCoercion */
+                $builder = self::makeBuilder($instrumentation, 'PDO::__construct', $function, $class, $filename, $lineno)
+                    ->setSpanKind(SpanKind::KIND_CLIENT)
+                    ->setAttribute(TraceAttributes::DB_CONNECTION_STRING, $params[0] ?? 'unknown')
+                    ->setAttribute(TraceAttributes::DB_USER, $params[1] ?? 'unknown')
+                    ->setAttribute(TraceAttributes::DB_SYSTEM, self::getDB($params[0]));
+                $parent = Context::getCurrent();
+                $span = $builder->startSpan();
+                Context::storage()->attach($span->storeInContext($parent));
+            },
+            post: static function (\PDO $pdo, array $params, mixed $statement, ?Throwable $exception) {
+                self::end($exception);            }
+        );
+
+        hook(
+            \PDO::class,
+            'query',
+            pre: static function (\PDO $pdo, array $params, string $class, string $function, ?string $filename, ?int $lineno) use ($instrumentation) {
+                /** @psalm-suppress ArgumentTypeCoercion */
+                $builder = self::makeBuilder($instrumentation, 'PDO::query', $function, $class, $filename, $lineno)
+                    ->setSpanKind(SpanKind::KIND_CLIENT)
+                    ->setAttribute(TraceAttributes::DB_STATEMENT, $params[0] ?? 'undefined');
+                $parent = Context::getCurrent();
+                $span = $builder->startSpan();
+                Context::storage()->attach($span->storeInContext($parent));
+            },
+            post: static function (\PDO $pdo, array $params, mixed $statement, ?Throwable $exception) {
+                self::end($exception);            }
+        );
+
+        hook(
+            \PDO::class,
+            'exec',
+            pre: static function (\PDO $pdo, array $params, string $class, string $function, ?string $filename, ?int $lineno) use ($instrumentation) {
+                /** @psalm-suppress ArgumentTypeCoercion */
+                $builder = self::makeBuilder($instrumentation, 'PDO::exec', $function, $class, $filename, $lineno)
+                    ->setSpanKind(SpanKind::KIND_CLIENT)
+                    ->setAttribute(TraceAttributes::DB_STATEMENT, $params[0] ?? 'undefined');
+                $parent = Context::getCurrent();
+                $span = $builder->startSpan();
+                Context::storage()->attach($span->storeInContext($parent));
+            },
+            post: static function (\PDO $pdo, array $params, mixed $statement, ?Throwable $exception) {
+                self::end($exception);            }
+        );
+
+        hook(
+            \PDOStatement::class,
+            'fetch',
```

Comment:
IMO we shouldn't create a span per `::fetch()` etc.; large result sets will lead to a lot of spans.

Reply:
I was thinking about and plan to change this behaviour in subsequent PR(s).